### ISSUE TITLE
[ADS]-Save Notebook option re-opens a new python session causing failure to [Reinstall notebook dependencies] 

### ISSUE THREAD

**Issue by pradm** [microsoft/azuredatastudio#7075]
Issue Type: <b>Bug</b>

It looks like Save Notebook functionality has issues in the latest ADS Insiders build; When a notebook is saved it automatically opens and loads the saved notebook in a new python session but that never opens a new
Notebook tab for a user to understand 2 notebooks are opened. When I close the Original Notebook it only closes the latest python session and the earlier session remains as it is. Is this a known issue which is being
fixed already. 

This is blocking my [Deploy SQL Server big data cluster] scenario using the ADS Wizard. The only workaround for now is to kill the python session which has opened a notebook in background and then move ahead
With the deployment. Can someone please take a look at this, I am able to reproduce this at will. I have filed a issues @ , In case if this is already known please close this issue as duplicate of the original reported issue.

Repro Steps:-
1.	Install latest ADS insider build which will installed version 1.11.0-insider (user setup)
2.	Post Installation open ADS
3.	Click on the 3dots and open [Deploy SQL Server big data cluster] option
4.	Select AKS and New AKS Environment
5.	Furnish the details and click on Open Notebook
6.	Run the First Cell it will fail with an exception and it requests the user to reinstall note book dependencies
7.	Try to run [Reinstall notebook dependencies] using command palette 
8.	The reinstall will fail with error [Cannot overwrite an existing Python installation while python is running. Please close any active notebooks before proceeding]
9.	To make sure our setting are saved in the Notebook – Save the Notebook to any location
10.	Close the Notebook
11.	Try to run [Reinstall notebook dependencies] using command palette 
12.	The execution will once again fail with [Cannot overwrite an existing Python installation while python is running. Please close any active notebooks before proceeding]
13.	Now we are stuck at this point and the only way out is to kill the Python process which has launched the notebook

Research Details:-
ADS Version
Version: 1.11.0-insider (user setup)
Commit: b8518f5795ce15a5a524cbc0f9e531b8e02d8456
Date: 2019-09-04T09:15:54.332Z
VS Code 1.37.0
Electron: 4.2.9
Chrome: 69.0.3497.128
Node.js: 10.11.0
V8: 6.9.427.31-electron.0
OS: Windows_NT x64 10.0.18362

Wizard Options Provided
![image](https://user-images.githubusercontent.com/27516588/64356783-0f421d00-d021-11e9-8bc8-dc18b9526cc4.png)

This opens up deploy-bdc-aks.ipynb  and while Execution of Check Dependencies Cell throws this error

An exception has occurred, use %tb to see the full traceback.

SystemExit: Please upgrade the Notebook dependency before you can proceed, you can do it by running the "Reinstall Notebook dependencies" command in command palette (View menu -> Command Palette…).
c:\users\pradm\azuredatastudio-python\0.0.1\lib\site-packages\IPython\core\interactiveshell.py:2969: UserWarning: To exit: use 'exit', 'quit', or Ctrl-D.
  warn("To exit: use 'exit', 'quit', or Ctrl-D.", stacklevel=1)

When I try to run [Reinstall Notebook dependencies] it fails with the following error
Cannot overwrite an existing Python installation while python is running. Please close any active notebooks before proceeding. 
![image](https://user-images.githubusercontent.com/27516588/64356879-3698ea00-d021-11e9-967f-3c4438610602.png)

I saved the deploy-bdc-aks.ipynb to my documents and closed it and executed the run [Reinstall Notebook dependencies] again it still reports [Cannot overwrite an existing Python installation while python is running. Please close any active notebooks before proceeding]

Output Window Details

... Creating C:\Users\pradm\AppData\Roaming\jupyter\instances\3d243132-5bff-49ec-b5bb-3e25b08f26b0
... Creating C:\Users\pradm\AppData\Roaming\jupyter\instances\3d243132-5bff-49ec-b5bb-3e25b08f26b0\config
... Creating C:\Users\pradm\AppData\Roaming\jupyter\instances\3d243132-5bff-49ec-b5bb-3e25b08f26b0\data
... Creating C:\Users\pradm\AppData\Roaming\jupyter\instances\3d243132-5bff-49ec-b5bb-3e25b08f26b0\config\custom
... Starting Notebook server
    > "C:\Users\pradm\azuredatastudio-python\0.0.1\python.exe" -m jupyter notebook --no-browser --no-mathjax --notebook-dir "C:\Users\pradm" --port=9628 --NotebookApp.token=da1ff06d93e0619b2757fbc3799be973e6924b251201e3e8
... Jupyter is running at http://localhost:9628/?token%3Dda1ff06d93e0619b2757fbc3799be973e6924b251201e3e8

... Creating C:\Users\pradm\AppData\Roaming\jupyter\instances\b65d6a53-8280-493f-be6c-708bd43e427e
... Creating C:\Users\pradm\AppData\Roaming\jupyter\instances\b65d6a53-8280-493f-be6c-708bd43e427e\config
... Creating C:\Users\pradm\AppData\Roaming\jupyter\instances\b65d6a53-8280-493f-be6c-708bd43e427e\data
... Creating C:\Users\pradm\AppData\Roaming\jupyter\instances\b65d6a53-8280-493f-be6c-708bd43e427e\config\custom
... Starting Notebook server
    > "C:\Users\pradm\azuredatastudio-python\0.0.1\python.exe" -m jupyter notebook --no-browser --no-mathjax --notebook-dir "c:\Users\pradm\Documents" --port=9256 --NotebookApp.token=695106d5cc2384ff837653d332e462f7f5a8ec919579389c
... Jupyter is running at http://localhost:9256/?token%3D695106d5cc2384ff837653d332e462f7f5a8ec919579389c

> "C:\Users\pradm\azuredatastudio-python\0.0.1\python.exe" -m jupyter notebook stop 9256
    stdout: Shutting down server on port 9256 ...
    stdout: 

Process Monitor Details
![image](https://user-images.githubusercontent.com/27516588/64356903-46183300-d021-11e9-83b5-49f52b04b89d.png)
![image](https://user-images.githubusercontent.com/27516588/64356915-4c0e1400-d021-11e9-81a2-189ff643a1a9.png)

Azure Data Studio version: azuredatastudio-insiders 1.11.0-insider (b8518f5795ce15a5a524cbc0f9e531b8e02d8456, 2019-09-04T09:15:54.332Z)
OS version: Windows_NT x64 10.0.18362

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i7-6600U CPU @ 2.60GHz (4 x 2808)|
|GPU Status|2d_canvas: enabled<br>flash_3d: enabled<br>flash_stage3d: enabled<br>flash_stage3d_baseline: enabled<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>native_gpu_memory_buffers: disabled_software<br>oop_rasterization: disabled_off<br>protected_video_decode: unavailable_off<br>rasterization: enabled<br>skia_deferred_display_list: disabled_off<br>skia_renderer: disabled_off<br>surface_synchronization: enabled_on<br>video_decode: enabled<br>viz_display_compositor: disabled_off<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|undefined|
|Memory (System)|15.83GB (8.72GB free)|
|Process Argv||
|Screen Reader|no|
|VM|0%|
</details>Extensions: none

**Comment by MaddyDev** [comment]
@alanrenmsft Did you come across this?

**Comment by alanrenmsft** [comment]
@MaddyDev I've see the error before, but closing the open notebooks will do the job.

**Comment by alanrenmsft** [comment]
@corivera are we relying on closing of the notebook to properly terminate the jupyter + python processes?

**Comment by corivera** [comment]
Not sure. @chlafreniere ?

**Comment by chlafreniere** [comment]
Yes. Notebooks need to be closed for the jupyter/python processes to be terminated.
